libs/sdk-ui-kit/src/Dialog/ShareDialog/types.ts

```diff
@@ -55,7 +56,63 @@ export interface IShareDialogProps {
     onApply: (payload: ISharingApplyPayload) => void;
     onCancel: () => void;
     onError?: (error: GoodDataSdkError) => void;
+    onInteraction?: (data: ShareDialogInteractionData) => void;
     isLockingSupported: boolean;
     labels: IShareDialogLabels;
     currentUserPermissions: CurrentUserPermissions;
 }
+
+// TODO: this will be fined tuned according to UX and PO requirements for action types
+/**
+ * @internal
+ */
+export type ShareDialogInteractionType =
```

Comment:
Seems a lot of prefixing :) Maybe it would worth it to think if we can split it to some subgroups or other type other than union.

Reply:
updated and reduced